bundles/binding/org.openhab.binding.powermax/src/main/java/org/openhab/binding/powermax/internal/state/PowerMaxPanelType.java

```diff
@@ -0,0 +1,196 @@
+        return label;
+    }
+
+    /**
+     * @return the number of managed partitions
+     */
+    public int getPartitions() {
+        return partitions;
+    }
+
+    /**
+     * @return the number of events stored in the event log
+     */
+    public int getEvents() {
+        return events;
+    }
+
+    /**
+     * @return the number of managed keyfobs
+     */
+    public int getKeyfobs() {
+        return keyfobs;
+    }
+
+    /**
+     * @return the number of managed uni-directional keypads
+     */
+    public int getKeypads1w() {
+        return keypads1w;
+    }
+
+    /**
+     * @return the number of managed bi-directional keypads
+     */
+    public int getKeypads2w() {
+        return keypads2w;
+    }
+
+    /**
+     * @return the number of managed sirens
+     */
+    public int getSirens() {
+        return sirens;
+    }
+
+    /**
+     * @return the number of managed user codes
+     */
+    public int getUserCodes() {
+        return userCodes;
+    }
+
+    public int getProntags() {
+        return prontags;
+    }
+
+    /**
+     * @return the number of managed wireless zones
+     */
+    public int getWireless() {
+        return wireless;
+    }
+
+    /**
+     * @return the number of managed wired zones
+     */
+    public int getWired() {
+        return wired;
+    }
+
+    /**
+     * @return the number of zones that can be customized by the user
+     */
+    public int getCustomZones() {
+        return customZones;
+    }
+
+    /**
+     * @return true is the panel is a PowerMaster panel type
+     */
+    public boolean isPowerMaster() {
+        return this == PowerMaxPanelType.POWERMASTER_10 || this == PowerMaxPanelType.POWERMASTER_30;
+    }
+
+    /**
+     * Get the ENUM value from its code number
+     *
+     * @param panelCode
+     *            the code stored by the panel
+     *
+     * @return the corresponding ENUM value
+     *
+     * @throws IllegalArgumentException
+     *             if no ENUM value corresponds to this code
+     */
+    public static PowerMaxPanelType fromCode(byte panelCode) {
+        for (PowerMaxPanelType panelType : PowerMaxPanelType.values()) {
+            if (panelType.getCode() == panelCode) {
+                return panelType;
+            }
+        }
+
+        throw new IllegalArgumentException("Invalid code: " + panelCode);
+    }
+
+    /**
+     * Get the ENUM value from its label
+     *
+     * @param label
+     *            the label
+     *
+     * @return the corresponding ENUM value
+     *
+     * @throws IllegalArgumentException
+     *             if no ENUM value corresponds to this label
+     */
+    public static PowerMaxPanelType fromLabel(String label) {
+        for (PowerMaxPanelType panelType : PowerMaxPanelType.values()) {
+            if (panelType.getLabel().equalsIgnoreCase(label)) {
+                return panelType;
+            }
+        }
+
+        throw new IllegalArgumentException("Invalid label: " + label);
+    }
```

Comment:
Side note: every `enum` has a `valueOf(String)` static method that would return the instance, so if you named your enum members using the labels, this method wouldn't be needed.